ice.c

```diff
@@ -3145,6 +3150,48 @@ void *janus_ice_send_thread(void *data) {
 			}
 			video_rtcp_last_sr = now;
 		}
+		/* Should we clean up old NACK buffers? */
+		if(now-last_nack_cleanup >= 500000) {
+			if(handle->audio_stream && handle->audio_stream->rtp_component) {
+				janus_ice_component *component = handle->audio_stream->rtp_component;
+				janus_mutex_lock(&component->mutex);
+				if(component->retransmit_buffer) {
+					GList *first = g_list_first(component->retransmit_buffer);
+					janus_rtp_packet *p = (janus_rtp_packet *)first->data;
+					while(p && (now - p->created >= max_nack_queue*G_USEC_PER_SEC)) {
+						/* Packet is too old, get rid of it */
+						first->data = NULL;
+						component->retransmit_buffer = g_list_delete_link(component->retransmit_buffer, first);
+						g_free(p->data);
+						p->data = NULL;
+						g_free(p);
+						first = g_list_first(component->retransmit_buffer);
+						p = (janus_rtp_packet *)(first ? first->data : NULL);
+					}
+				}
+				janus_mutex_unlock(&component->mutex);
+			}
+			if(handle->video_stream && handle->video_stream->rtp_component) {
+				janus_ice_component *component = handle->video_stream->rtp_component;
+				janus_mutex_lock(&component->mutex);
```

Comment:
Notice that starting from this line, the block of code til the end of this scope is identical to the block above. You can make a function for this, used something like this:

```c
janus_cleanup_nack_buffer(handle->video_stream->rtp_component);
```